server/aesdsocket.c

```diff
@@ -11,50 +12,288 @@
 #include <signal.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
+#include <pthread.h>
+#include <time.h>
+#include <stdbool.h>
+#include <errno.h>
 
 #define PORT_NUM "9000"
 #define MAX_BACKLOG 10
 #define OUTPUT_FILE_PATH "/var/tmp/aesdsocketdata"
-#define HUNDRED_BYTES 100
+#define MAX_BUF 100
 
 // global variables
-int socket_num;
-int client_fd;
-int file_fd;
+int socket_num; // fd for socket
+int client_fd; // fd for most recent thread connection
+int file_fd; // fd for output file
+
+struct sockaddr_in client_addr; // needed for IP address
+bool run_flag = true; // flag for main loop
+pthread_mutex_t mutex; // used for synchronization
```

Comment:
@bjornhnelson think about a more descriptive name for this.  What is it locking?